src/main/java/com/gargoylesoftware/htmlunit/javascript/host/xml/XMLHttpRequest.java

```diff
@@ -239,26 +245,21 @@ private void setState(final int state, Context context) {
 
             final Function onLoad = getOnload();
             if (onLoad != null) {
-                jsEngine.callFunction(containingPage_, onLoad, onLoad.getParentScope(), this, params);
+                jsEngine.callFunction(containingPage_, onLoad, onLoad.getParentScope(), this, paramOnLoad);
             }
 
-            List<Scriptable> handlers = getEventListenersContainer().getListeners(Event.TYPE_LOAD, false);
-            if (handlers != null) {
-                for (final Scriptable scriptable : handlers) {
-                    if (scriptable instanceof Function) {
-                        final Function function = (Function) scriptable;
-                        jsEngine.callFunction(containingPage_, function, function.getParentScope(), this, params);
-                    }
-                }
-            }
+            triggerJavascriptHandlers(getEventListenersContainer().getListeners(Event.TYPE_LOAD, false), paramOnLoad);
+            triggerJavascriptHandlers(getEventListenersContainer().getListeners(Event.TYPE_LOAD, false), paramOnLoad);
```

Comment:
i guess this has to be (Event.TYPE_LOAD, true)

if yes can we have a test case that proves this.

Reply:
Yes you're correct. Stupid copy & paste error.
I'll try to write a test for this.

I'd also assume that the following javadoc in EventListenersContainer is outdated by now? But will try to confirm.
```
    /**
     * Adds an event listener.
     *
     * @param type the event type to listen for (like "load")
     * @param listener the event listener
     * @param useCapture If {@code true}, indicates that the user wishes to initiate capture (not yet implemented)
     * @return {@code true} if the listener has been added
     */
    public boolean addEventListener(final String type, final Scriptable listener, final boolean useCapture) {
```